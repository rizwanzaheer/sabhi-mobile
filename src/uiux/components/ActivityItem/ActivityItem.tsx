import * as React from 'react'
import { ScrollView } from 'react-native'
import Container from '../Container/Container'
import Button, { ButtonBlocks } from '../Button/Button'
import Avatar from '../Avatar/Avatar'
import ActivityItemHeader from '../../components/ActivityItemHeader/ActivityItemHeader'
import * as Kancha from '../../types'
import { withTheme } from '../../theming'
import { BrandOptions } from '../../constants'

interface ActivityItemProps {
  /**
   * The unique id or message hash
   */
  id: string

  /**
   * The message type
   */
  type: 'w3c.vp' | 'w3c.vc' | 'sdr' | string

  /**
   * The timestamp for when this message was recieved or sent
   */
  date: number

  /**
   * The issuer of this message item
   */
  sender: Kancha.Identity

  /**
   * The activity that is takaing place
   */
  activity?: string

  /**
   * The subject
   */
  receiver: Kancha.Identity

  /**
   * The viewer
   */
  viewer: Kancha.Identity

  /**
   * The reason for the message
   */
  reason?: string

  /**
   * Message attachments
   */
  attachments?: any[]

  /**
   * Render attachment item
   */
  renderAttachment?: (attachmentItem: any, itemIndex: number) => React.ReactNode

  /**
   * Props to apply to attchment scrollview
   */
  attachmentScrollViewProps?: any

  /**
   * Message actions
   */
  actions?: string[]

  /*
   * The confirm action
   */
  confirm?: () => void

  /*
   * The reject action
   */
  reject?: () => void

  /**
   * Profile actions like being able to navigate to a profile
   */
  profileAction: (id: string) => void

  /**
   * Credential style
   */
  credentialStyle?: any
  /**
   * Theme prop
   */
  theme: any
}

const ActivityItem: React.FC<ActivityItemProps> = ({
  type,
  activity,
  sender,
  receiver,
  viewer,
  reason,
  attachments,
  attachmentScrollViewProps,
  renderAttachment,
  actions,
  profileAction,
  date,
  confirm,
  reject,
  theme,
}) => {
  const issProfileSource = sender.profileImage ? { source: { uri: sender.profileImage } } : {}

  return (
    <Container background={'primary'} marginBottom={10}>
      <Container flex={1} flexDirection={'row'} padding>
        <Container alignItems={'center'}>
          <Container>
            <Avatar
              {...issProfileSource}
              type={'circle'}
              gravatarType={'retro'}
              address={sender.did}
              size={40}
            />
          </Container>
        </Container>
        <Container marginLeft paddingRight flex={1}>
          <ActivityItemHeader
            viewer={viewer}
            issuer={sender}
            subject={receiver}
            profileAction={profileAction}
            date={date}
            activity={activity || theme.activity.messages[type]}
            reason={reason}
          />
          {actions && type === 'sdr' && viewer.did !== sender.did && (
            <Container flex={1} marginTop flexDirection={'row'}>
              <Container flex={2} marginRight={5}>
                {confirm && actions[0] && (
                  <Button
                    small
                    buttonText={actions[0]}
                    type={BrandOptions.Primary}
                    block={ButtonBlocks.Filled}
                    onPress={confirm}
                  ></Button>
                )}
              </Container>
              <Container flex={2}>
                {reject && actions[1] && (
                  <Button
                    small
                    buttonText={actions[1]}
                    type={BrandOptions.Secondary}
                    block={ButtonBlocks.Filled}
                    onPress={reject}
                  ></Button>
                )}
              </Container>
            </Container>
          )}
        </Container>
      </Container>
      {attachments && attachments.length > 0 && renderAttachment && (
        <Container flex={1}>
          <ScrollView
            horizontal
            style={{ flex: 1 }}
            showsHorizontalScrollIndicator={false}
            {...attachmentScrollViewProps}
          >
            {attachments.map((item: any, itemIndex: number) => {
              return renderAttachment(item, itemIndex)
            })}
          </ScrollView>
        </Container>
      )}
    </Container>
  )
}

export default withTheme(ActivityItem)
