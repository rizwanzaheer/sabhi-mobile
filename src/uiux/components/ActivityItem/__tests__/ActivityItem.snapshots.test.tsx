import * as React from 'react'
import { render, fireEvent } from 'react-native-testing-library'
import ActivityItem from '../ActivityItem'

const ATTACHMENT_1 = {
  hash: '01',
  jwt: '01',
  exp: 123445678910,
  sub: { shortId: 'Test Subject 1', did: '0x1efghssesd', profileImage: '' },
  iss: { shortId: 'Test Issuer 1', did: '0x1efgh', profileImage: '' },
  fields: [
    {
      type: 'name',
      value: 'Test name',
    },
    {
      type: 'phone',
      value: 'Test phone',
    },
  ],
}

const ATTACHMENT_2 = {
  hash: '01',
  jwt: '01',
  sub: { shortId: 'Test Subject 2', did: '0x1efghssesdss', profileImage: '' },
  iss: { shortId: 'Test Issuer 2', did: '0x1efghssss', profileImage: '' },
  fields: [
    {
      type: 'name',
      value: 'Test name',
    },
    {
      type: 'phone',
      value: 'Test phone',
    },
  ],
}

describe('Component(snapshots): ActivityItem', () => {
  const viewer = {
    did: 'ethr:did:123456876',
    profileImage: 'http://',
    shortId: 'Test Viewer',
  }
  const baseProps = {
    id: 'ZFGHFSJD',
    date: 123445678910,
    type: 'w3c.vc',
    sender: {
      did: 'ethr:did:123456',
      profileImage: 'http://',
      shortId: 'Test Issuer',
    },
    receiver: {
      did: 'ethr:did:123456',
      profileImage: 'http://',
      shortId: 'Test Subject',
    },
    activity: 'testing the components',
    profileAction: () => {},
  }

  it('should render with default required props', () => {
    const tree = render(<ActivityItem viewer={viewer} {...baseProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render with attachments', () => {
    const attachments = [ATTACHMENT_1, ATTACHMENT_2]

    const tree = render(<ActivityItem viewer={viewer} attachments={attachments} {...baseProps} />)
    expect(tree.toJSON).toMatchSnapshot()
  })

  it('should render with actions', () => {
    const actions = ['Confirm', 'Reject']
    const confirm = jest.fn()
    const reject = jest.fn()

    const tree = render(
      <ActivityItem
        {...baseProps}
        type={'sdr'}
        viewer={viewer}
        actions={actions}
        confirm={confirm}
        reject={reject}
      />,
    )
    expect(tree.toJSON).toMatchSnapshot()

    fireEvent.press(tree.getByText(/Confirm/i))
    expect(confirm).toHaveBeenCalled()

    fireEvent.press(tree.getByText(/Reject/i))
    expect(reject).toHaveBeenCalled()
  })
})
