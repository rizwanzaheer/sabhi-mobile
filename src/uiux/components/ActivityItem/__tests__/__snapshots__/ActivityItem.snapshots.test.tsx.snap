// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component(snapshots): ActivityItem should render with actions 1`] = `[Function]`;

exports[`Component(snapshots): ActivityItem should render with attachments 1`] = `[Function]`;

exports[`Component(snapshots): ActivityItem should render with default required props 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "alignItems": undefined,
      "backgroundColor": "#FFFFFF",
      "borderRadius": undefined,
      "flex": undefined,
      "flexDirection": undefined,
      "height": undefined,
      "justifyContent": undefined,
      "margin": undefined,
      "marginBottom": 10,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "width": undefined,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "flex": 1,
        "flexDirection": "row",
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": 15,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "alignItems": undefined,
            "backgroundColor": undefined,
            "borderRadius": undefined,
            "flex": undefined,
            "flexDirection": undefined,
            "height": undefined,
            "justifyContent": undefined,
            "margin": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "width": undefined,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "http://",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "#0078ff",
                "borderRadius": 20,
                "height": 40,
                "width": 40,
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        />
      </View>
    </View>
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": 1,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": 15,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": 15,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "alignItems": undefined,
            "backgroundColor": undefined,
            "borderRadius": undefined,
            "flex": undefined,
            "flexDirection": undefined,
            "height": undefined,
            "justifyContent": undefined,
            "margin": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "width": undefined,
          }
        }
      >
        <View
          pointerEvents="auto"
          style={
            Object {
              "alignItems": undefined,
              "backgroundColor": undefined,
              "borderRadius": undefined,
              "flex": 1,
              "flexDirection": undefined,
              "height": undefined,
              "justifyContent": undefined,
              "margin": undefined,
              "marginBottom": 4,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": undefined,
              "padding": undefined,
              "paddingBottom": undefined,
              "paddingHorizontal": undefined,
              "paddingLeft": undefined,
              "paddingRight": undefined,
              "paddingTop": undefined,
              "width": undefined,
            }
          }
        >
          <Text
            style={
              Object {
                "flexWrap": "wrap",
              }
            }
          >
            <Text
              onPress={[Function]}
              style={
                Object {
                  "color": "#333333",
                  "flexWrap": "wrap",
                  "fontSize": 17,
                  "fontWeight": "bold",
                }
              }
            >
              Test Issuer
            </Text>
            <Text
              style={
                Object {
                  "color": "#333333",
                  "flexWrap": "wrap",
                  "fontSize": 17,
                }
              }
            >
               
              testing the components
            </Text>
            <Text
              onPress={[Function]}
              style={
                Object {
                  "color": "#333333",
                  "flexWrap": "wrap",
                  "fontSize": 17,
                  "fontWeight": "bold",
                }
              }
            >
               
              Test Subject
            </Text>
          </Text>
        </View>
        <Text
          style={
            Object {
              "color": "#BBBBBB",
              "flexWrap": "wrap",
              "fontSize": 15,
            }
          }
        >
          over 46 years
           ago
        </Text>
      </View>
    </View>
  </View>
</View>
`;
